src/main/java/brave/opentracing/BraveSpan.java

```diff
@@ -25,7 +25,7 @@
  * <p>This type also includes hooks to integrate with the underlying {@linkplain brave.Tracer}.
  * Ex you can access the underlying span with {@link #unwrap}
  */
-public final class BraveSpan implements Span {
```

Comment:
mistake? I don't see anything trying to extend BraveSpan..

ps I would make whatever subtype of basespan might eventually happen package-private and also prefer package private extension vs protected as the concrete ActiveSpan and Span would be in the same package as BaseSpan.

This sort of advice is mainly from effective java and permeates most of Brave code. Doing the same things here makes it easier to reason with, and prevents us from breaking users who use accidentally exposed things or accidentally not-final types.